src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -69,171 +69,203 @@ bool MutationReplaceBinaryOperator::IsValidReplacementOperator(
-      new_function_name += "MulAssign";
+      function_name += "MulAssign";
       break;
     case clang::BinaryOperatorKind::BO_OrAssign:
-      new_function_name += "OrAssign";
+      function_name += "OrAssign";
       break;
     case clang::BinaryOperatorKind::BO_RemAssign:
-      new_function_name += "RemAssign";
+      function_name += "RemAssign";
       break;
     case clang::BinaryOperatorKind::BO_ShlAssign:
-      new_function_name += "ShlAssign";
+      function_name += "ShlAssign";
       break;
     case clang::BinaryOperatorKind::BO_ShrAssign:
-      new_function_name += "ShrAssign";
+      function_name += "ShrAssign";
       break;
     case clang::BinaryOperatorKind::BO_SubAssign:
-      new_function_name += "SubAssign";
+      function_name += "SubAssign";
       break;
     case clang::BinaryOperatorKind::BO_XorAssign:
-      new_function_name += "XorAssign";
+      function_name += "XorAssign";
       break;
     case clang::BinaryOperatorKind::BO_And:
-      new_function_name += "And";
+      function_name += "And";
       break;
     case clang::BinaryOperatorKind::BO_Or:
-      new_function_name += "Or";
+      function_name += "Or";
       break;
     case clang::BinaryOperatorKind::BO_Xor:
-      new_function_name += "Xor";
+      function_name += "Xor";
       break;
     case clang::BinaryOperatorKind::BO_LAnd:
-      new_function_name += "LAnd";
+      function_name += "LAnd";
       break;
     case clang::BinaryOperatorKind::BO_LOr:
-      new_function_name += "LOr";
+      function_name += "LOr";
       break;
     case clang::BinaryOperatorKind::BO_EQ:
-      new_function_name += "EQ";
+      function_name += "EQ";
       break;
     case clang::BinaryOperatorKind::BO_GE:
-      new_function_name += "GE";
+      function_name += "GE";
       break;
     case clang::BinaryOperatorKind::BO_GT:
-      new_function_name += "GT";
+      function_name += "GT";
       break;
     case clang::BinaryOperatorKind::BO_LE:
-      new_function_name += "LE";
+      function_name += "LE";
       break;
     case clang::BinaryOperatorKind::BO_LT:
-      new_function_name += "LT";
+      function_name += "LT";
       break;
     case clang::BinaryOperatorKind::BO_NE:
-      new_function_name += "NE";
+      function_name += "NE";
       break;
     case clang::BinaryOperatorKind::BO_Shl:
-      new_function_name += "Shl";
+      function_name += "Shl";
       break;
     case clang::BinaryOperatorKind::BO_Shr:
-      new_function_name += "Shr";
+      function_name += "Shr";
       break;
     default:
       assert(false && "Unsupported opcode");
   }
 
+  // To avoid problems of ambiguous function calls, the argument types (ignoring
+  // whether they are references or not) are baked into the mutation function
+  // name. Some type names have space in them (e.g. 'unsigned int'); such spaces
+  // are replaced with underscores.
+  function_name +=
+      "_" + SpaceToUnderscore(lhs_type) + "_" + SpaceToUnderscore(rhs_type);
+
```

Comment:
My above point is that I would move the logic for stringifying lhs and rhs types to here and just do it as part of tacking them on to the function name.

Reply:
Done.